.github/workflows/cd.yml

```diff
@@ -216,47 +212,54 @@ jobs:
           git commit -m "chore: replace sideloading placeholders"
 
       - name: update cli ai key
-        if: ${{ github.event_name == 'workflow_dispatch' && ( github.ref == 'refs/heads/main' || github.ref == 'refs/heads/prerelease' ) }}
+        if: ${{ github.event_name == 'workflow_dispatch' && github.event.inputs.preid != 'alpha' }}
         uses: jossef/action-set-json-field@v1
         with:
           file: ./packages/cli/package.json
           field: aiKey
           value: ${{ secrets.CLI_PUBLIC_AIKEY }}
 
       - name: update extension ai key
-        if: ${{ github.event_name == 'workflow_dispatch' && ( github.ref == 'refs/heads/main' || github.ref == 'refs/heads/prerelease' ) }}
+        if: ${{ github.event_name == 'workflow_dispatch' && github.event.inputs.preid != 'alpha' }}
         uses: jossef/action-set-json-field@v1
         with:
           file: ./packages/vscode-extension/package.json
           field: aiKey
           value: ${{ secrets.EXT_PUBLIC_AIKEY }}
 
       - name: commit change on local
-        if: ${{ github.event_name == 'workflow_dispatch' && ( github.ref == 'refs/heads/main' || github.ref == 'refs/heads/prerelease' ) }}
+        if: ${{ github.event_name == 'workflow_dispatch' && github.event.inputs.preid != 'alpha' }}
         run: |
           git add ./packages/cli/package.json ./packages/vscode-extension/package.json
           git commit -m "chore: update ai key"
 
       - name: publish alpha release to npm org
-        if: ${{ github.ref == 'refs/heads/dev' || github.ref == 'refs/heads/prerelease' }}
+        if: ${{ github.ref == 'refs/heads/dev'}}
```

Comment:
@wenytang-ms would you please confirm that the changes related to CD pipeline won't affect the release of VS 17.6 Preview 3?

Reply:
Should merge this PR after VS released. @1yefuwang1